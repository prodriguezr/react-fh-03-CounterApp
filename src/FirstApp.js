const FirstApp = () => {
  const greeting = 'Hello, World!';

  return (
    <>
      <h1>{greeting}</h1>
      <p>My first application</p>
    </>
  );
};

export default FirstApp;
