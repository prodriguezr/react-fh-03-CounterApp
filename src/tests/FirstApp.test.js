import { shallow } from 'enzyme';
import FirstApp from '../FirstApp';

describe('Tests on FirstApp', () => {
  test('should display the "<FirstApp/>" component correctly', () => {
    const greeting = 'Hola, soy Ironman';

    const wrapper = shallow(<FirstApp greeting={greeting} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should display subtitle from props', () => {
    const greeting = 'Hola, soy Ironman';
    const subtitle = 'Este es un subtitulo';

    const wrapper = shallow(
      <FirstApp greeting={greeting} subtitle={subtitle} />
    );

    expect(wrapper.find('p').text()).toBe(subtitle);
  });
});
