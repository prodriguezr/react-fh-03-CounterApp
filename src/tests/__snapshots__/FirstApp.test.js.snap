// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on FirstApp should display the "<FirstApp/>" component correctly 1`] = `
<Fragment>
  <h1>
    Hola, soy Ironman
  </h1>
  <p>
    Subtitle
  </p>
</Fragment>
`;
